contracts/epochs/FlowEpoch.cdc

```diff
@@ -647,6 +648,22 @@ pub contract FlowEpoch {
         return array
     }
 
+    /// Gets the the address of the secondary account that is associated
+    /// with collector and consensus nodes
+    pub fun updateMachineAccountForNode(qcVoterReference: &FlowEpochClusterQC.Voter?, dkgParticipantReference: &FlowDKG.Participant?) {
```

Comment:
This approach feels like we are mixing concerns a bit to me. I wonder if it would make more sense to do the machine account house-keeping in the user's own staking account (does this fit into how the staking collection works?).

The notion of a machine account is outside the scope of what the epoch smart contract needs to be aware of in my opinion. The epoch smart contract exposes the DKG and QC participant resources, but I don't think it should make assumptions about how those resources are stored or used. Flow Port put the resources into machine accounts with a 1-1 relation between node IDs and resources as the default case, but ultimately they are owned by node operators who can deal with them however they like.

Reply:
That is a good point. Though I thought we said that we wanted the machine account thing to be something that the protocol expects all node operators to do?

Reply:
Though I guess we can't enforce that since we are setting it in a transaction instead of a function of the contract, so probably shouldn't have it here

Reply:
> we said that we wanted the machine account thing to be something that the protocol expects all node operators to do

All node operators will eventually require a machine account that contains resources, but how those accounts and resources are set up can't really (and shouldn't) be enforced in this contract like you mentioned.